tests/Unit/Subscriber/Addons/CloudflareSubscriber/TestAutoPurgeByUrl.php

```diff
@@ -95,11 +99,14 @@ public function testShouldAutoPurgeByUrlWhenCacheEverything() {
 		Functions\when( 'current_user_can' )->justReturn( true );
 		Functions\when( 'get_current_user_id' )->justReturn( 1 );
 		Functions\when( 'is_wp_error' )->justReturn( false );
+		Functions\expect( 'get_rocket_i18n_home_url' )->once()->with( '' )->andReturn( 'http://example.org/' );
+		Functions\expect( 'get_feed_link' )->twice()->andReturn( 'http://example.org/feed/', 'http://example.org/feed/comments' );
 
-		$wp_error   = \Mockery::mock( \WP_Error::class );
 		$cloudflare = \Mockery::mock( \WP_Rocket\Addons\Cloudflare\Cloudflare::class );
-		$cloudflare->shouldReceive('has_page_rule')->andReturn( true );
-		$cloudflare->shouldReceive('purge_by_url')->andReturn( true );
+		$cloudflare->shouldReceive('has_page_rule')->with( 'cache_everything' )->andReturn( true );
+		$cloudflare->shouldReceive('purge_by_url')
+		           ->with( 1, [ '/hello-world', 'http://example.org/', 'http://example.org/feed/', 'http://example.org/feed/comments' ], '' )
```

Comment:
@crystinutzaa I added argument checks to further validate the behavior. Notice how we can test what we expect to be in `$purge_urls` using this approach.

Reply:
@hellofromtonya Indeed the test is way much clear and `$purge_urls` can be tested.